portfolio/src/main/java/com/google/sps/servlets/favBirdDataServlet.java

```diff
@@ -29,21 +35,42 @@
 @WebServlet("/favorite-bird")
 public class favBirdDataServlet extends HttpServlet {
 
-  private Map<String, Integer> birdVotes = new HashMap<>();
-
+  /**
+  * Fetches data with a query, sorts data into a hash map, 
+  * and translates to JSON to load to the graph
+  */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      
+    //create and prepare a query
+    Query query = new Query("Bird");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    
+    // store each vote in a hash map
+    Map<String, Integer> birdVotes = new HashMap<>();
+    for (Entity entity : results.asIterable()) {
+        String bird = (String) entity.getProperty("birdName");
+        int currentVotes = birdVotes.containsKey(bird) ? birdVotes.get(bird) : 0;
+        birdVotes.put(bird, currentVotes + 1);
+    }
+
+    // translate to json
     response.setContentType("application/json");
     Gson gson = new Gson();
     String json = gson.toJson(birdVotes);
     response.getWriter().println(json);
   }
 
+  /** Stores bird votes in Datastore using entities */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String bird = request.getParameter("bird");
-    int currentVotes = birdVotes.containsKey(bird) ? birdVotes.get(bird) : 0;
-    birdVotes.put(bird, currentVotes + 1);
+
+    Entity birdEntity = new Entity("Bird");
+    birdEntity.setProperty("birdName", request.getParameter("bird"));
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(birdEntity);
```

Comment:
is it possible to make the entity also store the score? it would be nice to only have one entry in the database per bird name and we can continue to make updates to that bird in the database. This would require setting a unique key per bird in the database and then retrieving that bird, adding 1 to the score, then updating its entry in the database in this function. While this requires one extra call to the db, it saves on the number of records being stored which might save more resources overall. (plus it'll help you simplify your logic in the get function)